compendium/DeclarativeServices/src/SCRAsyncWorkService.cpp

```diff
@@ -129,11 +136,12 @@ SCRAsyncWorkService::AddingService(
       if (newService) {
         std::atomic_store(&asyncWorkService, newService);
       }
-    } catch (...) {
-      std::string msg = "An exception occured while trying to get the "
-                        "user-provided implement of AsyncWorkService. "
-                        "Defaulting to the provided fallback "
-                        "solution";
+    } catch (const std::exception& ex) {
+      std::string msg = "An exception with the message \"";
+      msg += ex.what();
+      msg += "\" was caught while retrieving an instance of "
+             "cppmicroservices::async::detail::AsyncWorkService. Falling "
+             "back to the default.";
       logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_WARNING,
                   msg);
```

Comment:
I suggest using the `Log` function overload that takes a `std::exception_ptr` and capturing the exception.